### PR TITLE
Send customized memo to QBO 

### PR DESCRIPTION
**Pull request by Sravanksk**

### CODE REVIEW

File: apps/quickbooks_online/models.py [hunk 1]
```diff
@@ -35,16 +35,38 @@ def get_expense_purpose(workspace_id, lineitem, category) -> str:
     cluster_domain = fyle_connector.get_cluster_domain()
     org_id = Workspace.objects.get(id=workspace_id).fyle_org_id
 
+    employee_email = lineitem.employee_email
+    merchant = lineitem.vendor if lineitem.vendor else ''
```
Comment (Shwetabhk) [inline apps/quickbooks_online/models.py:39]: A lot of places we're doing this if condition and setting to empty sting
lets make a function out of this

File: apps/quickbooks_online/models.py [hunk 2]
```diff
@@ -35,16 +35,38 @@ def get_expense_purpose(workspace_id, lineitem, category) -> str:
...
+        elif field == 'category':
+            memo += category + ' - '
+        elif field == 'report_number':
+            memo += report_number + ' - '
+        elif field == 'spent_on':
+            memo += spent_at + ' - '
+        elif field == 'expense_link':
+            memo += expense_link + ' - '        
+
+    memo = memo[:-3] if memo[-1] == ' ' else memo
```
Comment (Shwetabhk) [inline apps/quickbooks_online/models.py:68]: This seems hacky, your for loop should take care of this

File: apps/workspaces/models.py [hunk 3]
```diff
@@ -65,6 +71,10 @@ class WorkspaceGeneralSettings(models.Model):
         base_field=models.CharField(max_length=100), default=get_default_chart_of_accounts,
         help_text='list of chart of account types to be imported into Fyle'
     )
+    customized_memo = ArrayField(
```
Comment (Shwetabhk) [inline apps/workspaces/models.py:74]: rename to `memo_structure`

Comment (Shwetabhk) [reply apps/workspaces/models.py:74]: Make sure and check that the correct order gets saved everytime

File: apps/workspaces/utils.py [hunk 4]
```diff
@@ -117,6 +117,8 @@ def create_or_update_general_settings(general_settings_payload: Dict, workspace_
             'import_tax_codes': general_settings_payload['import_tax_codes'],
             'change_accounting_period': general_settings_payload['change_accounting_period'],
             'charts_of_accounts': general_settings_payload['charts_of_accounts'],
+            'customized_memo': general_settings_payload['customized_memo'] if
```
Comment (Shwetabhk) [inline apps/workspaces/utils.py:120]: This is not required, db model will automatically call the default function

File: apps/workspaces/views.py [hunk 5]
```diff
@@ -428,3 +428,18 @@ def get(self, request, *args, **kwargs):
...
+        """
+        workspace_general_settings_object = WorkspaceGeneralSettings.objects.get(workspace_id=kwargs['workspace_id'])
+        serializer = WorkSpaceGeneralSettingsSerializer(
+            workspace_general_settings_object, data=request.data, partial=True
+        )
+        if serializer.is_valid():
+            serializer.save()
+            return Response(
+                data=serializer.data,
+                status=status.HTTP_201_CREATED
```
Comment (Shwetabhk) [inline apps/workspaces/views.py:444]: use `200` instead of `201`

File: apps/quickbooks_online/models.py [hunk 6]
```diff
@@ -35,16 +35,38 @@ def get_expense_purpose(workspace_id, lineitem, category) -> str:
...
         cluster_domain['cluster_domain'], lineitem.expense_id, org_id
     )
 
-    expense_purpose = ' purpose - {0}'.format(lineitem.purpose) if lineitem.purpose else ''
-    spent_at = ' spent on {0} '.format(lineitem.spent_at.date()) if lineitem.spent_at else ''
-    merchant = ' spent on merchant {0}'.format(lineitem.vendor) if lineitem.vendor else ''
-    return 'Expense by {0}{1} against category {2}{3} with report number - {4} -{5} - {6}'.format(
-        lineitem.employee_email, merchant, category, spent_at, lineitem.claim_number, expense_purpose,
-        expense_link)
+    workspace_general_settings = WorkspaceGeneralSettings.objects.get(workspace_id=workspace_id)
```
Comment (Shwetabhk) [inline apps/quickbooks_online/models.py:47]: pass general settings from the task.py function to the model function to this function

File: apps/quickbooks_online/models.py [hunk 7]
```diff
@@ -35,16 +35,38 @@ def get_expense_purpose(workspace_id, lineitem, category) -> str:
...
+    customized_memo = workspace_general_settings.customized_memo
+
+    memo = ''
+
+    for field in customized_memo:
```
Comment (Shwetabhk) [inline apps/quickbooks_online/models.py:52]: customized memo to be changed to memo_structure everywhere

**Review by Shwetabhk — CHANGES_REQUESTED** [review]

File: apps/workspaces/utils.py [hunk 8]
```diff
@@ -16,7 +16,7 @@
 from fyle_qbo_api.utils import assert_valid
 from .models import WorkspaceGeneralSettings
 from ..fyle.models import ExpenseGroupSettings
-
+from .models import get_default_memo_fields
```
Comment (ashwin1111) [inline apps/workspaces/utils.py:19]: Why is this required?

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**